sklearn/preprocessing/tests/test_data.py

```diff
@@ -2651,3 +2652,145 @@ def test_minmax_scaler_clip(feature_range):
+
+    for extrapolation in [None, True, 1, 'string']:
+        with pytest.raises(ValueError, match="extrapolation must be one of "
+                           "'error', 'constant', 'linear' or 'continue'."):
+            SplineTransformer(extrapolation=extrapolation).fit(X)
+
+    for include_bias in [None, 1, 'string']:
+        with pytest.raises(ValueError, match="include_bias must be bool."):
+            SplineTransformer(include_bias=include_bias).fit(X)
+
+    # Need at least 2 n_samples
+    with pytest.raises(ValueError):
+        SplineTransformer().fit([[1]])
+
+
+def test_spline_transformer_feature_names():
+    X = np.arange(20).reshape(10, 2)
+    splt = SplineTransformer(n_knots=3, degree=3, include_bias=True).fit(X)
+    feature_names = splt.get_feature_names()
+    assert_array_equal(feature_names,
+                       ['x0_sp_0', 'x0_sp_1', 'x0_sp_2', 'x0_sp_3', 'x0_sp_4',
+                        'x1_sp_0', 'x1_sp_1', 'x1_sp_2', 'x1_sp_3', 'x1_sp_4'])
+
+    splt = SplineTransformer(n_knots=3, degree=3, include_bias=False).fit(X)
+    feature_names = splt.get_feature_names(['a', 'b'])
+    assert_array_equal(feature_names,
+                       ['a_sp_0', 'a_sp_1', 'a_sp_2', 'a_sp_3',
+                        'b_sp_0', 'b_sp_1', 'b_sp_2', 'b_sp_3'])
+
+
+@pytest.mark.parametrize('degree', range(1, 5))
+@pytest.mark.parametrize('n_knots', range(3, 5))
+@pytest.mark.parametrize('knots', ['uniform', 'quantile'])
+def test_spline_transformer_unity_decomposition(degree, n_knots, knots):
+    # test that the splines are decomposition of unity,
+    # i.e. sum up to 1 per row, if we stay in between boundaries.
+    X = np.linspace(0, 1, 100)[:, None]
+    # make the boundaries 0 and 1 part of X_train, for sure.
+    X_train = np.r_[[[0]], X[::2, :], [[1]]]
+    X_test = X[1::2, :]
+    splt = SplineTransformer(n_knots=n_knots, degree=degree, knots=knots,
+                             include_bias=True)
+    splt.fit(X_train)
+    for X in [X_train, X_test]:
+        assert_allclose(np.sum(splt.transform(X), axis=1), 1)
+
+
+@pytest.mark.parametrize(['bias', 'intercept'],
+                         [(True, False), (False, True)])
+def test_spline_transformer_linear_regression(bias, intercept):
+    # test that the splines fit a sinusodial curve pretty well
+    X = np.linspace(0, 10, 100)[:, None]
+    y = np.sin(X[:, 0]) + 2  # +2 to avoid the value 0 in assert_allclose
+    pipe = Pipeline(
+        [['spline', SplineTransformer(n_knots=15, degree=3, include_bias=bias,
+                                      extrapolation='constant')],
+         ['ols', LinearRegression(fit_intercept=intercept)]])
+    pipe.fit(X, y)
+    assert_allclose(pipe.predict(X), y, rtol=1e-3)
+
+
+@pytest.mark.parametrize(['bias', 'intercept'],
+                         [(True, False), (False, True)])
+@pytest.mark.parametrize('degree', [1, 2, 3, 4, 5])
+def test_spline_transformer_extrapolation(bias, intercept, degree):
```

Comment:
Do we need to test the `fit_intercept` here?

Reply:
Yes. `predict` should give the same results for the combinations `include_bias=True, fit_intercept=False` and for `include_bias=False, fit_intercept=True`. As `include_bias=False` throws away the last basis spline which is  important for the extrapolation, i.e. not zero beyond the boundary.